### PR TITLE
Unix socket factory

### PR DESCRIPTION
**Pull request by semihalev**
I added new factory for unix sockets. It's useful for multi worker. I got 10% better performance vs default SocketFactory for unix socket. 

Example PHP Relay:
```
$relay = new Goridge\SocketRelay(
            '/tmp/go.sock.'.getmypid(),
            null,
            Goridge\SocketRelay::SOCK_UNIX
);
```

### CODE REVIEW

**Comment by wolfy-j** [comment]
Hi,

as far I can understand you are creating a new socket connection for every worker, I have couple issues with this PR:

1) Code reuse factory.ls for every new worker even when listeners are different (line 57). 
2) A socket connection is not being closed anywhere.
3) Direct association between socket and worker does not require same pid negotiation and channeling and socket factory (it has been done for multiplexing reasons only). So code must be simpler.
4) Tests + performance benchmarks.

**Comment by semihalev** [comment]
I understand your considers, I closed listener after first connection accepted because only one connection acceptable. I will add test and performance benchmarks when i have time. Also I found a small bug in worker and changed.

File: unix_socket_factory.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+		return nil, err
+	}
+
+	if err := w.start(); err != nil {
+		return nil, errors.Wrap(err, "process error")
+	}
+
+	socketFile := fmt.Sprintf("%s.%d", f.file, *w.Pid)
+
+	os.Remove(socketFile)
```
Comment (wolfy-j) [inline unix_socket_factory.go]: i would group this operation with line 51 not 54, but it's ok.

**Comment by wolfy-j** [comment]
Sorry for the delay, I have been on a business trip. Please see my comment.

**Comment by semihalev** [comment]
I added some tests and benchmarks.

These results for static pool with 8 workers;

Benchmark_UnixSocket_Pool_ExecEcho-8      	   20000	     62451 ns/op	     132 B/op	       8 allocs/op
Benchmark_Unix_Pool_ExecEcho-8            	   20000	     60845 ns/op	     132 B/op	       8 allocs/op

**Comment by wolfy-j** [comment]
a lot of your issues were addressed in new update (see #9), please update your PR or close it :)

**Comment by semihalev** [comment]
Woow too many changes, I close it. I will check this later.

**Comment by wolfy-j** [comment]
@semihalev Have you compared it to pipes performance? We are planning to have a set of performance optimizations to be done and I'm curious if we can use your approach.

**Pull request closed without merging**